test/functional/shada/registers_spec.lua

```diff
@@ -160,4 +160,16 @@ describe('ShaDa support code', function()
     reset()
     eq({{'\171'}, 'v'}, getreg('e'))
   end)
+
+  it('remembers which register was the unnamed register when loading',
+  function()
+    setreg('0', {'zero'}, 'c')
+    setreg('1', {'one'}, 'cu')
+    setreg('2', {'two'}, 'c')
+    nvim_command('qall')
+    reset()
+    eq({{'zero'}, 'v'}, getreg('0'))
+    eq({{'one'}, 'v'}, getreg('1'))
+    eq({{'one'}, 'v'}, getreg('"'))
+  end)
```

Comment:
If the unnamed register isn't set and nothing is stored in register `0`, it will be blank regardless of what the other registers are set to, just like Vim.